fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/customized/DatumWriterCustomization.java

```diff
@@ -0,0 +1,40 @@
+package com.linkedin.avro.fastserde.customized;
+
+/**
+ * Customization for serialization.
+ * This class is expandable.
+ */
+public class DatumWriterCustomization {
+  private CheckMapTypeFunction checkMapTypeFunction;
+
+  private DatumWriterCustomization(Builder builder) {
+    this.checkMapTypeFunction = builder.checkMapTypeFunction;
+  }
+
+  public CheckMapTypeFunction getCheckMapTypeFunction() {
+    return checkMapTypeFunction;
+  }
+
+  public interface CheckMapTypeFunction {
+    void apply(Object o);
+  }
```

Comment:
What is the goal of this function? To throw if the check fails?

Do we not have a use case where we want to transform the map (e.g. sort the keys) prior to serialization?

Reply:
yeah, in OSS Venice, there is a safeguard when serializing the write compute result to make sure each map field will use an ordered map.
https://github.com/linkedin/venice/blob/main/clients/da-vinci-client/src/main/java/com/linkedin/davinci/serializer/avro/MapOrderPreservingGenericDatumWriter.java#L23

Reply:
Ah ok, so we just fail there... I thought we were sorting on the fly... so I guess this is good enough to support our current use case. IDK if there would ever be a use case for on the fly transformation, so I guess we can leave it as is for now.

Reply:
Well, ordered map doesn't mean it is sorted.
`IndexedHashMap` is not sorted, and it is more like a LinkedList to preserve the original order.

Reply:
Actually, we do have a use case where we alter the written map, right here: https://github.com/linkedin/venice/blob/585036c9b0553c8a96653d9255c1e2dbd4bd93a3/internal/venice-client-common/src/main/java/org/apache/avro/generic/DeterministicMapOrderGenericDatumWriter.java#L46

Reply:
Will address this in another PR.